### PR TITLE
Version 5.1.0

### PR DESCRIPTION
**Pull request by vegaro**
## 5.1.0

- Updated error log message for cases when there are no products configured in the dashboard 
  - https://github.com/RevenueCat/purchases-android/pull/517
- Added new method `setMixpanelDistinctID` as a convenience method for setting the required attribute for the Mixpanel integration
  - https://github.com/RevenueCat/purchases-android/pull/518
- Adds missing docs for `purchaseProduct` and `purchasePackage`
  - https://github.com/RevenueCat/purchases-android/pull/520
- Added `setFirebaseAppInstanceID` to allow associating RevenueCat users with Firebase.
  - https://github.com/RevenueCat/purchases-android/pull/523
- Adds `syncObserverModeAmazonPurchase` to sync Amazon purchases with RevenueCat. To be used in observer mode integrations.
  - https://github.com/RevenueCat/purchases-android/pull/483
- Better error handling when Amazon prices are missing
  - https://github.com/RevenueCat/purchases-android/pull/527
- Better error handling when Amazon user data cannot be loaded
  - https://github.com/RevenueCat/purchases-android/pull/528
- Fix detecting RevenueCat errors when loading offerings
  - https://github.com/RevenueCat/purchases-android/pull/529
- Adds price and marketplace to the headers of the `/receipts` API call
  - https://github.com/RevenueCat/purchases-android/pull/529

[Full Changelog](https://github.com/revenuecat/purchases-android/compare/5.0.0...5.1.0)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,28 @@
+## 5.1.0
+
+## 5.1.0
```
Comment (aboedo) [inline CHANGELOG.md:3]: duplicated line

File: CHANGELOG.latest.md [hunk 2]
```diff
@@ -1,46 +1,22 @@
...
+- Added `setFirebaseAppInstanceID` to allow associating RevenueCat users with Firebase.
+  - https://github.com/RevenueCat/purchases-android/pull/523
+- Adds `syncObserverModeAmazonPurchase` to sync Amazon purchases with RevenueCat. To be used in observer mode integrations.
+  - https://github.com/RevenueCat/purchases-android/pull/483
+- Better error handling when Amazon prices are missing
+  - https://github.com/RevenueCat/purchases-android/pull/527
+- Better error handling when Amazon user data cannot be loaded
+  - https://github.com/RevenueCat/purchases-android/pull/528
+- Fix detecting RevenueCat errors when loading offerings
+  - https://github.com/RevenueCat/purchases-android/pull/529
```
Comment (aboedo) [inline CHANGELOG.latest.md:18]: Looks good! 

For iOS we've started separating API updates from other changes for the sake of readability (https://github.com/RevenueCat/purchases-ios/releases)
I'm not sure we'll be able to keep this once we also automate changelogs, though, but it's kinda nice

File: CHANGELOG.latest.md [hunk 3]
```diff
@@ -1,46 +1,22 @@
...
-    https://github.com/RevenueCat/purchases-android/pull/489
-- Updated to BillingClient 4.1
-    https://github.com/RevenueCat/purchases-android/pull/500
-- Prevent Amazon from swallowing exceptions
-    https://github.com/RevenueCat/purchases-android/pull/499
-- Improvements on Amazon implementation preventing a lot of issues related to price and currency parsing
-    https://github.com/RevenueCat/purchases-android/pull/503
-
-[Full Changelog](https://github.com/revenuecat/purchases-android/compare/main...5.0.0)
+## 5.1.0
```
Comment (aboedo) [inline CHANGELOG.latest.md:1]: I think this file usually doesn't have the version number, that's why we have the duplicated line

**Review by aboedo — CHANGES_REQUESTED** [review]
There's a duplicated line but good to go after that

File: CHANGELOG.latest.md (hunk 2 above)
Comment (vegaro) [reply CHANGELOG.latest.md:18]: I like it. I bet there's a way we can still automate it by tagging the type of change in the PR

File: CHANGELOG.md [hunk 4]
```diff
@@ -45,7 +73,7 @@ dependencies {
 - Improvements on Amazon implementation preventing a lot of issues related to price and currency parsing
     https://github.com/RevenueCat/purchases-android/pull/503
 
-[Full Changelog](https://github.com/revenuecat/purchases-android/compare/main...5.0.0)
+[Full Changelog](https://github.com/revenuecat/purchases-android/compare/5.0.0-rc3...5.0.0)
```
Comment (aboedo) [inline CHANGELOG.md:76]: good catch

**Review by aboedo — APPROVED** [review]

**Pull request merged**